src/lib/device-types/build-info-facade.ts

```diff
@@ -14,6 +15,34 @@ export const getIsIgnored = memoizee(
 	{ promise: true, preFetch: true, maxAge: BUILD_PROPERTY_CACHE_EXPIRATION },
 );
 
+export const getLogoUrl = memoizee(
+	async (
+		normalizedSlug: string,
+		buildId: string,
+	): Promise<string | undefined> => {
+		if (!FILES_HOST) {
+			return;
+		}
+		try {
+			const logoKey = getImageKey(normalizedSlug, buildId, 'logo.svg');
+			const hasLogo = await fileExists(logoKey);
+			if (!hasLogo) {
+				return;
+			}
+
+			const encodedLogoPath = getImageKey(
+				normalizedSlug,
+				encodeURIComponent(buildId),
```

Comment:
Why do you URL-encode here but not above? Also, why only the `buildId` and not the slug?

Reply:
> Why do you URL-encode here but not above?

AWS SDK allows us to pass plain string values and ignore url encoding, (probably that's done internally).
On the other hand since the logoUrl will be a url (:smiley:), I needed to do this since otherwise versions/`buildIds`s like `2.46.1+rev3.prod` would be parsed as `2.46.1 rev3.prod` on the files.balena-cloud.com side, which would return a 404.

> Also, why only the buildId and not the slug?

Atm only the `version` has the `+` issue, but indeed I should have used url encode on all of them :+1: